c_tests/tests/rel_path.c

```diff
@@ -0,0 +1,58 @@
+// Compiler:
+// Run-time:
+//   env-var: YK_PRINT_IR=1
+//   stderr:
+//   stderr:
+//     define internal void @__yk_compiled_trace_0(i32* %0) {
+//        store i32 2, i32* %0, align 4, !tbaa !0
+//        ret void
+//     }
+//     ...
+
+// Check that running a traced binary via a relative path works.
+//
+// FIXME An optimising compiler can remove all of the code between start/stop
+// tracing.
+
+#include <assert.h>
+#include <err.h>
+#include <libgen.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  if (*argv[0] == '/') {
+    // Reinvoke ourself with a relative path.
+    char *base = basename(argv[0]);
+    if (base == NULL)
+      err(EXIT_FAILURE, "basename");
+
+    char *dir = dirname(argv[0]);
+    if (dir == NULL)
+      err(EXIT_FAILURE, "dirname");
+    if (chdir(dir) != 0)
+      err(EXIT_FAILURE, "chdir");
+
+    if (execl(base, base, NULL) != 0)
```

Comment:
I think that this should be `if (execl(...) == -1` to be clear about the range of return codes?